engine/src/main/java/org/hibernate/search/query/dsl/impl/ConnectedMultiFieldsTermQueryBuilder.java

```diff
@@ -190,7 +192,15 @@ private Query createTermQuery(FieldContext fieldContext, String term) {
 		return query;
 	}
 
-	private List<String> getAllTermsFromText(String fieldName, String localText, Analyzer analyzer) {
+	private List<String> getAllTermsFromText(String fieldName, String localText, AnalyzerReference analyzerReference) {
+		Analyzer analyzer = null;
+		if ( analyzerReference.getAnalyzer() != null ) {
+			analyzer = analyzerReference.getAnalyzer();
+		}
+		else if ( analyzerReference.getRemote() != null ) {
+			return Arrays.asList( localText );
```

Comment:
This won't work. If someone is looking for "several words", it's going to create a term query with term "several words" which won't match anything as "several words" has probably been tokenized and separated in several terms by the analyzer configured on Elasticsearch. Unless you count on having defined an analyzer locally to tokenize the text.

I think for the EXACT case, we can probably use a multi_match query which is the best fit IMHO.

As for the FUZZY case, a multi_match with fuzziness should do the trick.

For wildcard queries, it's a bit more complicated as it's not that easy to find something equivalent in Elasticsearch. And, moreover both the following cases make perfect sense: the one where you want the words tokenized (typically for a search ) or where you don't want them tokenized (especially useful for autocompletion). So automatically tokenizing the search terms is not a solution either. The query string query allows too much features and you can't disable any of them. The simple query string query only allows for prefix queries, not wildcards. I think our only way to make this feature work in the same way with Elasticsearch is to call the _analyze API in this case and exploit the tokens returned.

Thoughts?

(I don't think this should delay this patch though)